core/templates/dev/head/services/csrf-token.service.ts

```diff
@@ -16,40 +16,35 @@
  * @fileoverview Service for managing CSRF tokens.
  */
 
-// This needs to be imported first instead of using the global definition
-// because Angular doesn't support global definitions and every library used
-// needs to be imported explicitly.
-import $ from 'jquery';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('CsrfTokenService', [function() {
-  var tokenPromise = null;
+@Injectable({
+  providedIn: 'root'
+})
+export class CsrfTokenService {
+  constructor(private httpClient: HttpClient) {}
+  tokenPromise = null;
 
-  return {
-    initializeToken: function() {
-      if (tokenPromise !== null) {
-        throw new Error('Token request has already been made');
-      }
-      // We use jQuery here instead of Angular's $http, since the latter creates
-      // a circular dependency.
-      tokenPromise = $.ajax({
-        url: '/csrfhandler',
-        type: 'GET',
-        dataType: 'text',
-        dataFilter: function(data) {
-          // Remove the protective XSSI (cross-site scripting inclusion) prefix.
-          var actualData = data.substring(5);
-          return JSON.parse(actualData);
-        },
-      }).then(function(response) {
+  initializeToken() {
+    if (this.tokenPromise !== null) {
+      throw new Error('Token request has already been made');
+    }
+
+    this.tokenPromise = this.httpClient.get('/csrfhandler').toPromise()
+      .then((response: any) => {
         return response.token;
       });
-    },
+  }
```

Comment:
Could we do this in the service constructor? (I'm not sure if that is the ideal solution and if it would actually work)